lib/OpenQA/Worker/Common.pm

```diff
@@ -368,8 +368,9 @@ sub call_websocket {
             }
             else {
                 $ws = undef;
-                if ($tx->completed) {
-                    call_websocket($ua_url->parse($tx->res->headers->location));
+                if (my $location_header = ($tx->completed ? $tx->res->headers->location : undef)) {
+                    print "Following ws redirection to: $location_header\n";
+                    call_websocket($ua_url->parse($location_header));
```

Comment:
And beside readability? I mean this changes the behavior to ensure that it only tries to follow a redirection if the header is actually present and not stupidly retry all the time.